src/Orleans/Core/GrainIdentity.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using Orleans.Runtime;
+
+namespace Orleans.Core
+{
+    internal class GrainIdentity : IGrainIdentity
```

Comment:
Making `GrainId` implement `IGrainIdentity` works. However, merging `GetGrainId()` and `GetGrainIdentity()` of `GrainExtensions` is not possible at this point. The reason is that the runtime is still relying on the `GrainId` for things such as observers and we currently have no way to inject the `GrainId` to a `Grain` during testing. In fact, even though `GrainId` now implements `IGrainIdentity`, we cannot assume that we can cast any `IGrainIdentity` to a `GrainId` because a test double is not a `GrainId` but implements `IGrainIdentity`. 

The source of the issue is that the runtime is talking directly to the `GrainId` and that class does a lot more that `IGrainIdentity`. We should refactor it (backlog or open an issue) to make sure that the runtime only talks to the `IGrainIdentity` to retrieve identity information; the rest of functionality that `GrainId` offers today should be refactored into more granular interfaces.